packages/node-core/src/utils/blockSizeBuffer.ts

```diff
@@ -0,0 +1,34 @@
+// Copyright 2020-2021 OnFinality Limited authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import { Queue } from './autoQueue';
+
+export class BlockSizeBuffer extends Queue<number> {
+  constructor(capacity: number) {
+    super(capacity);
+  }
+
+  putMany(items: number[]): void {
+    if (items.length > this.capacity) {
+      throw new Error(`Queue out of capacity`);
+    }
+
+    if (this.capacity && items.length > this.freeSpace) {
+      this.takeMany(items.length);
```

Comment:
Following on from the other comments. This is really implicit, it should be moved to where putMany is called